examples/fungible-token/ft/src/lib.rs

```diff
@@ -29,23 +29,44 @@ near_sdk::setup_alloc!();
 #[derive(BorshDeserialize, BorshSerialize, PanicOnDefault)]
 pub struct Contract {
     token: FungibleToken,
-    reference: String,
-    reference_hash: Base64VecU8,
+    metadata: FungibleTokenMetadata,
 }
 
 #[near_bindgen]
 impl Contract {
-    /// Initializes the contract with the given total supply owned by the given `owner_id`.
+    /// Initializes the contract with the given total supply owned by the given `owner_id` with
+    /// default metadata (for example purposes only).
+    #[init]
+    pub fn new_default_meta(owner_id: ValidAccountId, total_supply: U128) -> Self {
```

Comment:
What's the effect of multiple `#[init]` methods?  Can you only initialize with one of them?

Reply:
You can add multiple macros like this. I believe under the hood the main thing that's happening is the contract doesn't try to retrieve storage and fit it into the struct. Eugene has used this to do fancy contract upgrades by using this tactic.

Reply:
`init` is a marker for a constructor